src/Init/Data/BitVec/Lemmas.lean

```diff
@@ -651,6 +660,24 @@ theorem zero_shiftLeft (n : Nat) : 0#w <<< n = 0#w := by
   cases h₁ : decide (i < m) <;> cases h₂ : decide (n ≤ i) <;> cases h₃ : decide (i < n)
   all_goals { simp_all <;> omega }
 
+theorem shiftLeft_xor_distrib (x y : BitVec w) (n : Nat) :
+    (x ^^^ y) <<< n = (x <<< n) ^^^ (y <<< n) := by
+  ext i
+  simp
+  by_cases h : (i < n) <;> simp [h]
+
+theorem shiftLeft_and_distrib (x y : BitVec w) (n : Nat) :
+    (x &&& y) <<< n = (x <<< n) &&& (y <<< n) := by
+  ext i
+  simp
+  by_cases h : (i < n) <;> simp [h]
+
+theorem shiftLeft_or_distrib (x y : BitVec w) (n : Nat) :
+    (x ||| y) <<< n = (x <<< n) ||| (y <<< n) := by
+  ext i
+  simp
+  by_cases h : (i < n) <;> simp [h]
```

Comment:
```suggestion
  by_cases h : i < n <;> simp [h]
```

A minort stylistic nitpick, but are these parentheses nessecary? Such parentheses are not present on line 565, but are present on line 667 and 673. It's probably best to be consistent.

Reply:
Resolved.